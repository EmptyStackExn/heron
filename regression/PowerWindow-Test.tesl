// Power window
// http://wdi.supelec.fr/software/ModHelX/PowerWindow

unit-clock btn_up       // the button is pulled up
unit-clock btn_neutral  // the button is released
unit-clock up           // the TFSM receives an up event
unit-clock stop         // the TFSM receives a stop event
unit-clock power        // the TFSM produces a power event
unit-clock update_power // the SDF model receives a new power command
unit-clock react        // the SDF model reacts to its inputs
rational-clock realtime // real-time in seconds
rational-clock bus      // time scale of the CAN bus

// The transmission delay on the CAN bus is 2 ms
tag relation realtime = 0.002 * bus
btn_up time delayed by 1.0 on bus implies up
btn_neutral time delayed by 1.0 on bus implies stop

// When the TFSM receives an input, it updates its output instantaneously
up implies power
stop implies power

// The transmission delay on the CAN bus is 2 ms
power time delayed by 1.0 on bus implies update_power

// The window must react every 50ms
react time delayed by 0.05 on realtime implies react

// Scenario
rational-clock react_scenario sporadic 0.0
tag relation react_scenario = 0.001 * realtime
react_scenario implies react

rational-clock up_scenario sporadic 5.0
tag relation up_scenario = 1000.0 * realtime
up_scenario implies btn_up

rational-clock neutral_scenario sporadic 320.0
tag relation neutral_scenario = 1000.0 * realtime
neutral_scenario implies btn_neutral

// Output options
// @tagref realtime
@dumpres
@maxstep 14
@policy asap
@run
// @output tikz standalone xscale 20 select btn_up, btn_neutral, up, stop, power, update_power, react
@output tikz select btn_up btn_neutral up stop power update_power react realtime bus react_scenario up_scenario neutral_scenario