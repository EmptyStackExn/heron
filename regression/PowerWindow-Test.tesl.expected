  \node       at (1+0.1, -2+0.1)  {\Cross} ;
  \node       at (1+0.1, -3+0.1)  {\Cross} ;
  \node       at (1+0.1, -10+0.1)  {\Cross} ;
  \node       at (1+0.1, -11+0.1)  {\Cross} ;
  \node       at (1+0.1, 0+0.1)  {\Cross} ;
  \node       at (1+0.1, -1+0.1)  {\Cross} ;
  \node       at (1+0.1, -4+0.1)  {\Cross} ;
  \node       at (2+0.1, -2+0.1)  {\Cross} ;
  \node       at (2+0.1, -3+0.1)  {\Cross} ;
  \node       at (2+0.1, -9+0.1)  {\Cross} ;
  \node       at (2+0.1, -11+0.1)  {\Cross} ;
  \node       at (2+0.1, -1+0.1)  {\Cross} ;
  \node       at (2+0.1, -4+0.1)  {\Cross} ;
  \node       at (2+0.1, -6+0.1)  {\Cross} ;
  \node       at (3+0.1, -3+0.1)  {\Cross} ;
  \node       at (3+0.1, -9+0.1)  {\Cross} ;
  \node       at (3+0.1, -10+0.1)  {\Cross} ;
  \node       at (3+0.1, -11+0.1)  {\Cross} ;
  \node       at (3+0.1, 0+0.1)  {\Cross} ;
  \node       at (3+0.1, -1+0.1)  {\Cross} ;
  \node       at (3+0.1, -6+0.1)  {\Cross} ;
  \node       at (4+0.1, -2+0.1)  {\Cross} ;
  \node       at (4+0.1, -3+0.1)  {\Cross} ;
  \node       at (4+0.1, -9+0.1)  {\Cross} ;
  \node       at (4+0.1, -10+0.1)  {\Cross} ;
  \node       at (4+0.1, -11+0.1)  {\Cross} ;
  \node       at (4+0.1, 0+0.1)  {\Cross} ;
  \node       at (4+0.1, -1+0.1)  {\Cross} ;
  \node       at (4+0.1, -4+0.1)  {\Cross} ;
  \node       at (4+0.1, -6+0.1)  {\Cross} ;
  \node       at (5+0.1, -2+0.1)  {\Cross} ;
  \node       at (5+0.1, -3+0.1)  {\Cross} ;
  \node       at (5+0.1, -9+0.1)  {\Cross} ;
  \node       at (5+0.1, -10+0.1)  {\Cross} ;
  \node       at (5+0.1, -11+0.1)  {\Cross} ;
  \node       at (5+0.1, 0+0.1)  {\Cross} ;
  \node       at (5+0.1, -1+0.1)  {\Cross} ;
  \node       at (5+0.1, -4+0.1)  {\Cross} ;
  \node       at (6+0.1, -2+0.1)  {\Cross} ;
  \node       at (6+0.1, -3+0.1)  {\Cross} ;
  \node       at (6+0.1, -9+0.1)  {\Cross} ;
  \node       at (6+0.1, -10+0.1)  {\Cross} ;
  \node       at (6+0.1, -11+0.1)  {\Cross} ;
  \node       at (6+0.1, 0+0.1)  {\Cross} ;
  \node       at (6+0.1, -1+0.1)  {\Cross} ;
  \node       at (6+0.1, -4+0.1)  {\Cross} ;
  \node       at (7+0.1, -2+0.1)  {\Cross} ;
  \node       at (7+0.1, -3+0.1)  {\Cross} ;
  \node       at (7+0.1, -9+0.1)  {\Cross} ;
  \node       at (7+0.1, -10+0.1)  {\Cross} ;
  \node       at (7+0.1, -11+0.1)  {\Cross} ;
  \node       at (7+0.1, 0+0.1)  {\Cross} ;
  \node       at (7+0.1, -1+0.1)  {\Cross} ;
  \node       at (7+0.1, -4+0.1)  {\Cross} ;
  \node       at (8+0.1, -2+0.1)  {\Cross} ;
  \node       at (8+0.1, -3+0.1)  {\Cross} ;
  \node       at (8+0.1, -9+0.1)  {\Cross} ;
  \node       at (8+0.1, -10+0.1)  {\Cross} ;
  \node       at (8+0.1, -11+0.1)  {\Cross} ;
  \node       at (8+0.1, 0+0.1)  {\Cross} ;
  \node       at (8+0.1, -1+0.1)  {\Cross} ;
  \node       at (8+0.1, -4+0.1)  {\Cross} ;
  \node       at (9+0.1, -2+0.1)  {\Cross} ;
  \node       at (9+0.1, -3+0.1)  {\Cross} ;
  \node       at (9+0.1, -9+0.1)  {\Cross} ;
  \node       at (9+0.1, -10+0.1)  {\Cross} ;
  \node       at (9+0.1, -11+0.1)  {\Cross} ;
  \node       at (9+0.1, 0+0.1)  {\Cross} ;
  \node       at (9+0.1, -1+0.1)  {\Cross} ;
  \node       at (9+0.1, -4+0.1)  {\Cross} ;
  \node       at (10+0.1, -2+0.1)  {\Cross} ;
  \node       at (10+0.1, -3+0.1)  {\Cross} ;
  \node       at (10+0.1, -9+0.1)  {\Cross} ;
  \node       at (10+0.1, -10+0.1)  {\Cross} ;
  \node       at (10+0.1, -11+0.1)  {\Cross} ;
  \node       at (10+0.1, 0+0.1)  {\Cross} ;
  \node       at (10+0.1, -1+0.1)  {\Cross} ;
  \node       at (10+0.1, -4+0.1)  {\Cross} ;
  \node       at (11+0.1, -2+0.1)  {\Cross} ;
  \node       at (11+0.1, -3+0.1)  {\Cross} ;
  \node       at (11+0.1, -9+0.1)  {\Cross} ;
  \node       at (11+0.1, -10+0.1)  {\Cross} ;
  \node       at (11+0.1, 0+0.1)  {\Cross} ;
  \node       at (11+0.1, -4+0.1)  {\Cross} ;
  \node       at (11+0.1, -6+0.1)  {\Cross} ;
  \node       at (12+0.1, -2+0.1)  {\Cross} ;
  \node       at (12+0.1, -9+0.1)  {\Cross} ;
  \node       at (12+0.1, -10+0.1)  {\Cross} ;
  \node       at (12+0.1, -11+0.1)  {\Cross} ;
  \node       at (12+0.1, 0+0.1)  {\Cross} ;
  \node       at (12+0.1, -1+0.1)  {\Cross} ;
  \node       at (12+0.1, -6+0.1)  {\Cross} ;
  \node       at (13+0.1, -2+0.1)  {\Cross} ;
  \node       at (13+0.1, -3+0.1)  {\Cross} ;
  \node       at (13+0.1, -9+0.1)  {\Cross} ;
  \node       at (13+0.1, -10+0.1)  {\Cross} ;
  \node       at (13+0.1, -11+0.1)  {\Cross} ;
  \node       at (13+0.1, 0+0.1)  {\Cross} ;
  \node       at (13+0.1, -1+0.1)  {\Cross} ;
  \node       at (13+0.1, -4+0.1)  {\Cross} ;
  \node       at (13+0.1, -6+0.1)  {\Cross} ;
  \node       at (14+0.1, -2+0.1)  {\Cross} ;
  \node       at (14+0.1, -3+0.1)  {\Cross} ;
  \node       at (14+0.1, -9+0.1)  {\Cross} ;
  \node       at (14+0.1, -10+0.1)  {\Cross} ;
  \node       at (14+0.1, -11+0.1)  {\Cross} ;
  \node       at (14+0.1, 0+0.1)  {\Cross} ;
  \node       at (14+0.1, -1+0.1)  {\Cross} ;
  \node       at (14+0.1, -4+0.1)  {\Cross} ;
  \node[tick] at (1, -9)  {} ;
  \node[date] at (1, -9)  {0.} ;
  \node[date] at (1, -7)  {0.} ;
  \node[date] at (1, -8)  {0.} ;
  \node[tick] at (1, -6)  {} ;
  \node[date] at (1, -10)  {0.} ;
  \node[date] at (1, -11)  {0.} ;
  \node[tick] at (2, -10)  {} ;
  \node[date] at (2, -10)  {5.} ;
  \node[date] at (2, -7)  {5E-3} ;
  \node[date] at (2, -8)  {2.5} ;
  \node[date] at (2, -9)  {5E-6} ;
  \node[tick] at (2, 0)  {} ;
  \node[date] at (2, -11)  {5.} ;
  \node[date] at (3, -8)  {3.5} ;
  \node[tick] at (3, -2)  {} ;
  \node[date] at (3, -7)  {7E-3} ;
  \node[tick] at (3, -4)  {} ;
  \node[date] at (3, -9)  {7E-6} ;
  \node[date] at (3, -10)  {7.} ;
  \node[date] at (3, -11)  {7.} ;
  \node[date] at (4, -8)  {4.5} ;
  \node[tick] at (4, -5)  {} ;
  \node[date] at (4, -7)  {9E-3} ;
  \node[date] at (4, -9)  {9E-6} ;
  \node[date] at (4, -10)  {9.} ;
  \node[date] at (4, -11)  {9.} ;
  \node[date] at (5, -7)  {0.05} ;
  \node[tick] at (5, -6)  {} ;
  \node[date] at (5, -8)  {25.} ;
  \node[date] at (5, -9)  {5E-5} ;
  \node[date] at (5, -10)  {50.} ;
  \node[date] at (5, -11)  {50.} ;
  \node[date] at (6, -7)  {0.1} ;
  \node[tick] at (6, -6)  {} ;
  \node[date] at (6, -8)  {50.} ;
  \node[date] at (6, -9)  {1E-4} ;
  \node[date] at (6, -10)  {100.} ;
  \node[date] at (6, -11)  {100.} ;
  \node[date] at (7, -7)  {0.15} ;
  \node[tick] at (7, -6)  {} ;
  \node[date] at (7, -8)  {75.} ;
  \node[date] at (7, -9)  {1.5E-4} ;
  \node[date] at (7, -10)  {150.} ;
  \node[date] at (7, -11)  {150.} ;
  \node[date] at (8, -7)  {0.2} ;
  \node[tick] at (8, -6)  {} ;
  \node[date] at (8, -8)  {100.} ;
  \node[date] at (8, -9)  {2E-4} ;
  \node[date] at (8, -10)  {200.} ;
  \node[date] at (8, -11)  {200.} ;
  \node[date] at (9, -7)  {0.25} ;
  \node[tick] at (9, -6)  {} ;
  \node[date] at (9, -8)  {125.} ;
  \node[date] at (9, -9)  {2.5E-4} ;
  \node[date] at (9, -10)  {250.} ;
  \node[date] at (9, -11)  {250.} ;
  \node[date] at (10, -7)  {0.3} ;
  \node[tick] at (10, -6)  {} ;
  \node[date] at (10, -8)  {150.} ;
  \node[date] at (10, -9)  {3E-4} ;
  \node[date] at (10, -10)  {300.} ;
  \node[date] at (10, -11)  {300.} ;
  \node[tick] at (11, -11)  {} ;
  \node[date] at (11, -11)  {320.} ;
  \node[date] at (11, -7)  {0.32} ;
  \node[date] at (11, -8)  {160.} ;
  \node[date] at (11, -9)  {3.2E-4} ;
  \node[date] at (11, -10)  {320.} ;
  \node[tick] at (11, -1)  {} ;
  \node[date] at (12, -8)  {161.} ;
  \node[tick] at (12, -3)  {} ;
  \node[date] at (12, -7)  {0.322} ;
  \node[tick] at (12, -4)  {} ;
  \node[date] at (12, -9)  {3.22E-4} ;
  \node[date] at (12, -10)  {322.} ;
  \node[date] at (12, -11)  {322.} ;
  \node[date] at (13, -8)  {162.} ;
  \node[tick] at (13, -5)  {} ;
  \node[date] at (13, -7)  {0.324} ;
  \node[date] at (13, -9)  {3.24E-4} ;
  \node[date] at (13, -10)  {324.} ;
  \node[date] at (13, -11)  {324.} ;
  \node[date] at (14, -7)  {0.35} ;
  \node[tick] at (14, -6)  {} ;
  \node[date] at (14, -8)  {175.} ;
  \node[date] at (14, -9)  {3.5E-4} ;
  \node[date] at (14, -10)  {350.} ;
  \node[date] at (14, -11)  {350.} ;
